### DISCUSSION TITLE
Requests chaining/dependency management

### DISCUSSION THREAD

**Discussion by 1ace** [hoppscotch/hoppscotch#2157] [Ideas]
My team is currently using Postman, but we're starting to have too many tests to reasonably run them sequentially.
We need to be able to run independent requests in parallel to keep execution time down, but Postman doesn't support that.

Hoppscotch seems like the best alternative, but efforts to implement this feature have [stalled](https://github.com/hoppscotch/hoppscotch/issues/218#issuecomment-583806117).

I'm not committing to anything, but I'll evaluate Hoppscotch as an alternative to Postman for our needs, and look at the code to see if I can give a shot at implementing this feature.
I've already implemented connection pools and topological sorting in other clients so these concepts are already familiar to me; the only difficulty I foresee is with the UI & UX: **how to get the user to simply & efficiently express all their needs?**

This discussion is here mostly so that anyone who's already taken a look at implementing this can provide hints, and potential users can tell us about their use-cases.

If anyone has experience (good or bad) with similar apps providing this feature, don't hesitate to share what should be done the same way, or (especially) what turns out to be a pain to do.

**Comment by 1ace** [comment]
As a start to the discussion, it seems that there are two things that we need when describing dependencies:
- that previous request needs to have completed (and its tests passed) before starting this one
- that previous request returned something that I need to use in this request (eg. resource ID)

**Comment by AndrewBastin** [comment]
This is a really interesting thingy going on here 🤣 

We could do something for sequential stuff (we have to come up with some mechanism for this but is viable).

But for parallel execution, I mean, barring the internal architecture concerns to accomodate this (and there is a lot to be honest 😅 ), the main issue for us to tackle this head on is implementing a proper UX for this. 

We want to keep the grace of the current UX in terms of how approachable it is, hence we are kinda trying to limit how dense each features are getting.

We also have been thinking of some UX paradigm shifts where the app is further split off into concerns or contexts (for eg., "Request Building", "Testing" etc.). Doing this can accomodate a good UX for this. But its still just in discussion phase and will take a lot of time to go through with.

At the end, why a lot of people use Hoppscotch over Postman (for the most part), is how lean it is. So this is a really important concern to us.

cc: @liyasthomas

**Comment by Behnke89** [comment]
Hi,

I just wanted to backup this Feature Request. Im considering using Hoppscotch for quite a while now. BUT we definitly need request chaining for Collections. Some applications are stateful and we do some integrative chained Tests. For me this would be the last piece to replace Postman.

**Comment by cjchand** [comment]
Chiming in with a "me too" on the need for this feature. This is used heavily by our end-to-end testing teams who aren't devs by trade. The ability to click a button and launch a functional test suite is a must-have for those users. CLI is not really a viable option for them.

**Comment by domgom** [comment]
Hi all, 
IMHO The ability to run folders and control the request flow in code (setNextRequest) are the foundation of functional test suits and a key aspect to a lot of QA teams.
I won't say that you need to plan your roadmap based on what your competitors do but I think there is a huge opportunity considering how postman has aggresively limited the local runs (from 25/mo on free version to 250/mo on enterprise version).
In a nutshell this is my case for considering hoppscotch and I'm sure more people will be hitting the same wall as these moneygrab limits get enforced.

**Comment by MightyGoldenJA** [comment]
Well, was kinda blown away that this feature is missing, I respectfully add my "me too" on the need for this feature.
